### PR TITLE
Update puppet-lint and puppet-syntax default ignore paths

### PR DESCRIPTION
**Pull request by DavidS**
These changes are to align the defaults more with what's currently in use
in the field, and improve puppet4 compatibility.

### CODE REVIEW

File: lib/puppetlabs_spec_helper/rake_tasks.rb [hunk 1]
```diff
@@ -429,14 +429,24 @@ def max_thread_limit
...
+  ]
 end
 
 require 'puppet-syntax/tasks/puppet-syntax'
 PuppetSyntax.exclude_paths ||= []
 PuppetSyntax.exclude_paths << "spec/fixtures/**/*"
 PuppetSyntax.exclude_paths << "pkg/**/*"
 PuppetSyntax.exclude_paths << "vendor/**/*"
+if Puppet.version.to_f < 4.0
+  PuppetSyntax.exclude_paths << "types/**/*"
```
Comment (bastelfreak) [inline lib/puppetlabs_spec_helper/rake_tasks.rb]: stupid question, why is this needed?

Comment (DavidS) [inline lib/puppetlabs_spec_helper/rake_tasks.rb]: with puppet<4 puppet syntax will choke on type declarations.

**Comment by bastelfreak** [comment]
LGTM!
